web-jehlomat/src/screens/SetNewPassword/SetNewPassword.tsx

```diff
@@ -0,0 +1,144 @@
+﻿import React, {FunctionComponent} from "react";
+import {useHistory} from "react-router-dom";
+import {useMediaQuery} from "@mui/material";
+import {Formik, FormikHelpers} from "formik";
+import {AxiosResponse} from "axios";
+import * as yup from "yup";
+import API from "config/baseURL";
+import {media} from "utils/media";
+import hasOwnProperty from "utils/hasOwnProperty";
+import SignInImage from "assets/images/sign-in.svg";
+import {Header} from "Components/Header/Header";
+import PrimaryButton from "Components/Buttons/PrimaryButton/PrimaryButton";
+import TwoColumns from "Components/Layout/TwoColumns";
+import {Input,Inputs,FormTitle,TransparentForm,Buttons,Illustration} from "./Components";
+
+const texts = {
+  TITLE: "Nové heslo",
+  MOBILE_TITLE: "Nové heslo",
+  INPUT__PASS__LABEL: "Nové heslo",
+  INPUT__PASS__PLACEHOLDER: "Snadno zapomatovatelné, těžce uhodnutelné",
+  INPUT__RE_PASS__LABEL: "Kontrola hesla",
+  INPUT__RE_PASS__PLACEHOLDER: "Zadejte stejné heslo",
+  VALIDATIONS__REQUIRED: "Povinná položka",
+  VALIDATIONS__SYMBOLS: "Heslo musí obsahovat číslo, velké a malé písmeno",
+  VALIDATIONS__MIN_LENGTH: "Heslo musí být 8 znaků dlouhé",
+  VALIDATIONS__MATCH: "Hesla musí být stejná",
+  BUTTONS__SEND: "Nastav heslo",
+};
+
+interface ResetPasswordFormProps {
+  password: string;
+  rePassword: string;
+}
+
+const init: ResetPasswordFormProps = {
+  password: '',
+  rePassword: '',
+}
+
+const schema = yup.object({
+  password: yup.string()
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])/, texts.VALIDATIONS__SYMBOLS)
+    .min(8, texts.VALIDATIONS__MIN_LENGTH)
+    .required(texts.VALIDATIONS__REQUIRED),
+  rePassword: yup.string().oneOf([yup.ref('password')], texts.VALIDATIONS__MATCH),
+});
+
+const ResetPassword: FunctionComponent = () => {
+  const isDesktop = useMediaQuery(media.gt('mobile'));
+  const history = useHistory();
+  
+  const handleSubmit = async (values: ResetPasswordFormProps, { setErrors }: FormikHelpers<ResetPasswordFormProps>) => {
+    try {
+      const response: AxiosResponse = await API.post('/api/v1/jehlomat/change-password', values);
+
+      const status = response.status;
+      const isSuccess = status >= 200 && status < 300;
+      
+      if (isSuccess) {
+        history.push('/');
+        return;
+      }
+
+      if (response.data && 
```

Comment:
what is the logic, if the condition is not truth?

Reply:
That's a great question and honestly don't know, I haven't seen any alert/toast service, so not sure how to communicate with users when something succeeded.
Is there something like that?

Reply:
I saw on LoginPage redirection to the error page only, otherwise there is no